action/login/login.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package login
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/go-vela/cli/internal/output"
+
+	"github.com/go-vela/sdk-go/vela"
+
+	"github.com/go-vela/types/library"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Config represents the configuration necessary
+// to perform login related requests with Vela.
+type Config struct {
+	Action   string
+	Username string
+	Password string
+	OTP      string
+	Retry    bool
+}
+
+// Login authenticates and logs in to Vela via the API based off the provided configuration.
+func (c *Config) Login(client *vela.Client) error {
```

Comment:
This is where we add the new `Login()` function.

The purpose of this function is to authenticate to Vela and initiate the login workflow.